src/metadata/MwdbIntegrate.py

```diff
@@ -0,0 +1,55 @@
+import logging
+
+import requests
+import urllib
+import mwdblib
+from mwdblib import Malwarecage
+from metadata import Metadata
+
+
+class MWDBIntegrateMetadata(Metadata):
+    __depends_on__ = ["CuckooBinariesMetadata"]
+
+    def __init__(self, mwdb_api_token: str, mwdb_api_url: str) -> None:
+        super().__init__()
+        self.token = mwdb_api_token
+        self.mwdb_url = mwdb_api_url
+
+    def extract(self, matched_fname, dependent_meta):
+        if not dependent_meta.get("cuckoo_hash"):
+            return {}
+
+        hash = dependent_meta.get("cuckoo_hash")["value"]
+
+        cached = self.cache_fetch(hash)
+        if cached:
+            return cached
+
+        if not self.token:
+            logging.error(
+                "MWDB metadata not fetched, MWDB_API_TOKEN variable was not set"
+            )
+            return {}
+
+        try:
+            api = mwdblib.MalwarecageAPI(
+                api_url=self.mwdb_url, api_key=self.token
+            )
+            mwdb = Malwarecage(api=api)
+            file = mwdb.query_file(hash)
+
+            job_id = dependent_meta["job"]
+            file.add_metakey("mquery", job_id)
+
+        except:
+            logging.exception("Failed to post mquery metadata")
+            return {}
+
+        obj = {}
+
+        obj["mwdb_sample"] = {
+            "display_text": "mwdb",
+            "url": f"http://mwdb.cert.pl/sample/{job_id}",
```

Comment:
Hardcoded URL for cert.pl's instance. Can it be configured for future open-sourcing of mwdb?

Reply:
I think the whole plugin system needs a revamp, though I'm not sure how to do it in an extensible way. But you're right, I missed that (api url above is configurable but the hyperlink is not).